src/components/Section.tsx

```diff
@@ -0,0 +1,195 @@
+  const [isEdited, setIsEdited] = useState(0);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEdited && inputRef.current) {
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 0);
+    }
+  }, [isEdited]);
+
+  const onDoubleClick = (id: number, title: string) => {
+    setIsEdited(id);
+    setEditedText(title);
+  };
+
+  const handleInputBlur = () => {
+    setIsEdited(0);
+  };
+
+  const handleInputFocus = (id: number) => {
+    setIsEdited(id);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedText(e.target.value);
+  };
+
+  async function handleSubmit(e: React.FormEvent, id: number) {
+    e.preventDefault();
+
+    if (editedText.trim()) {
+      setTodos((currentTodo) => currentTodo.map((todo) => {
+        return todo.id === id ? { ...todo, title: editedText } : todo;
+      }));
+
+      try {
+        await Promise.all(todos.map((todo) => {
+          if (todo.id === id) {
+            return postService.updateTodoTitle(id, { title: editedText });
+          }
+
+          return 'error';
+        }));
+      } catch (error) {
+        setErrorMessage(ErrorMesage.addingError);
+      }
+    } else {
+      setTodos((currentTodo) => currentTodo.filter((todo) => todo.id !== id));
+      try {
+        await Promise.all(todos.map((todo) => {
+          if (todo.id === id) {
+            return postService.deleteTodo(id);
+          }
+
+          return 'error';
+        }));
+      } catch (error) {
+        setErrorMessage(ErrorMesage.deletingError);
+      }
+    }
+
+    setIsEdited(0);
+  }
+
+  async function deleteData(idToDelete: number) {
+    try {
+      setTodos(currentTodos => currentTodos
+        .filter((currentTodo) => currentTodo.id !== idToDelete));
+      await postService.deleteTodo(idToDelete);
+    } catch (error) {
+      setErrorMessage(ErrorMesage.deletingError);
+    }
+  }
+
+  async function updateData(taskId: number, completed: boolean) {
+    setTodos((currentTodos) => (
+      currentTodos.map((currentTodo) => (
+        currentTodo.id === taskId ? (
+          { ...currentTodo, completed: !completed }
+        ) : (
+          currentTodo
+        )
+      ))
+    ));
+    try {
+      await postService.updateTodo(taskId, { completed: !completed });
+    } catch (error) {
+      setErrorMessage(ErrorMesage.deletingError);
+    }
+  }
+
+  let filteringTodos;
+
+  switch (tasksFilter) {
+    case TasksFilter.active:
+      filteringTodos = todos.filter((todo) => !todo.completed);
+      break;
+
+    case TasksFilter.completed:
+      filteringTodos = todos.filter((todo) => todo.completed);
+      break;
+    default:
+      filteringTodos = todos;
+  }
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {filteringTodos.map(({ title, id, completed }) => (
+        <div
+          data-cy="Todo"
+          className={cn('todo', { completed: completed === true })}
```

Comment:
```suggestion
          className={cn('todo', { completed: completed })}
```

Reply:
Can I stay this condition? If I do yours changes - I received mistake: 'Expected property shorthand'